src/compute/app_runner.rs

```diff
@@ -0,0 +1,210 @@
+    const ENV_IEXEC_TASK_ID: &str = "IEXEC_TASK_ID";
+    const ENV_SIGN_WORKER_ADDRESS: &str = "SIGN_WORKER_ADDRESS";
+    const ENV_SIGN_TEE_CHALLENGE_PRIVATE_KEY: &str = "SIGN_TEE_CHALLENGE_PRIVATE_KEY";
+    const ENV_WORKER_HOST: &str = "WORKER_HOST_ENV_VAR";
+    const DEFAULT_WORKER_HOST: &str = "localhost:8080";
+
+    #[test]
+    fn test_start_task_id_missing() {
+        temp_env::with_vars_unset(vec![ENV_IEXEC_TASK_ID], || {
+            assert_eq!(start(), 3, "Should return 3 if IEXEC_TASK_ID is missing");
+        });
+    }
+
+    #[test]
+    fn test_start_run_fails_get_challenge_fails_signer_address_missing() {
+        let env_vars_to_set = vec![
+            (ENV_IEXEC_TASK_ID, Some(CHAIN_TASK_ID)),
+            (
+                ENV_SIGN_TEE_CHALLENGE_PRIVATE_KEY,
+                Some(ENCLAVE_CHALLENGE_PRIVATE_KEY),
+            ),
+        ];
+        let env_vars_to_unset = vec![ENV_SIGN_WORKER_ADDRESS];
+
+        temp_env::with_vars(env_vars_to_set, || {
+            temp_env::with_vars_unset(env_vars_to_unset, || {
+                assert_eq!(
+                    start(),
+                    2,
+                    "Should return 2 if get_challenge fails due to missing signer address"
+                );
+            });
+        });
+    }
+
+    #[test]
+    fn test_start_run_fails_get_challenge_fails_private_key_missing() {
+        let env_vars_to_set = vec![
+            (ENV_IEXEC_TASK_ID, Some(CHAIN_TASK_ID)),
+            (ENV_SIGN_WORKER_ADDRESS, Some(WORKER_ADDRESS)),
+        ];
+        let env_vars_to_unset = vec![ENV_SIGN_TEE_CHALLENGE_PRIVATE_KEY];
+
+        temp_env::with_vars(env_vars_to_set, || {
+            temp_env::with_vars_unset(env_vars_to_unset, || {
+                assert_eq!(
+                    start(),
+                    2,
+                    "Should return 2 if get_challenge fails due to missing private key"
+                );
+            });
+        });
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_start_run_fails_get_challenge_succeeds_send_exit_cause_api_error() {
+        let mock_server = MockServer::start().await;
+
+        Mock::given(method("POST"))
+            .and(path(format!("/compute/pre/{}/exit", CHAIN_TASK_ID)))
+            .respond_with(ResponseTemplate::new(500))
+            .mount(&mock_server)
+            .await;
+
+        let mock_server_addr_string = mock_server.address().to_string();
+
+        let result_code = tokio::task::spawn_blocking(move || {
+            let env_vars = vec![
+                (ENV_IEXEC_TASK_ID, Some(CHAIN_TASK_ID)),
+                (ENV_SIGN_WORKER_ADDRESS, Some(WORKER_ADDRESS)),
+                (
+                    ENV_SIGN_TEE_CHALLENGE_PRIVATE_KEY,
+                    Some(ENCLAVE_CHALLENGE_PRIVATE_KEY),
+                ),
+                (ENV_WORKER_HOST, Some(mock_server_addr_string.as_str())),
+            ];
+
+            temp_env::with_vars(env_vars, || start())
```

Comment:
Clippy recommandation
```suggestion
            temp_env::with_vars(env_vars, start)
```